### PR TITLE
Allow grid slots to have attributes

### PR DESCRIPTION
**Pull request by Korko**

### CODE REVIEW

File: src/resources/views/components/mail/grid.blade.php [hunk 1]
```diff
@@ -8,21 +8,21 @@
                 <td align="center" valign="top">
                     <table {{ $attributes->merge(['class' => "container grid grid-col-$columns"]) }} width="600" cellpadding="0" cellspacing="0" border="0">
                         <tr>
-                            <td class="mobile" align="center" valign="top">
+                            <td class="mobile" align="center" valign="top" {{ $one?->attributes }}>
```
Comment (BJNSTNKVC) [inline src/resources/views/components/mail/grid.blade.php]: This wont work as the component will simply use the default class, which in this case is 'mobile'. In order for this to work, you would need to merge the default class with whatever is passed when calling a component:

```blade
<td {{ $one?->attributes->merge(['class' => 'mobile']) }} align="center" valign="top" >
    {{ $one ?: $slot }}
</td>
```

File: src/resources/views/components/mail/grid.blade.php [hunk 2]
```diff
@@ -8,21 +8,21 @@
...
                                 {{ $one ?: $slot }}
                             </td>
                             @if ($columns >= 2)
-                                <td class="mobile" align="center" valign="top">
+                                <td class="mobile" align="center" valign="top" {{ $two->attributes }}>
```
Comment (BJNSTNKVC) [inline src/resources/views/components/mail/grid.blade.php]: This wont work as the component will simply use the default class, which in this case is 'mobile'. In order for this to work, you would need to merge the default class with whatever is passed when calling a component:

```blade
<td {{ $two->attributes->merge(['class' => 'mobile']) }} align="center" valign="top" >
    {{ $two }}
</td>
```

File: src/resources/views/components/mail/grid.blade.php [hunk 3]
```diff
@@ -8,21 +8,21 @@
...
                                     {{ $two }}
                                 </td>
                             @endif
                             @if ($columns >= 3)
-                                <td class="mobile" align="center" valign="top">
+                                <td class="mobile" align="center" valign="top" {{ $three->attributes }}>
```
Comment (BJNSTNKVC) [inline src/resources/views/components/mail/grid.blade.php]: This wont work as the component will simply use the default class, which in this case is 'mobile'. In order for this to work, you would need to merge the default class with whatever is passed when calling a component:

```blade
<td {{ $three->attributes->merge(['class' => 'mobile']) }} align="center" valign="top" >
    {{ $three }}
</td>
```

File: src/resources/views/components/mail/grid.blade.php [hunk 4]
```diff
@@ -8,21 +8,21 @@
...
                                     {{ $three }}
                                 </td>
                             @endif
                             @if ($columns >= 4)
-                                <td class="mobile" align="center" valign="top">
+                                <td class="mobile" align="center" valign="top" {{ $four->attributes }}>
```
Comment (BJNSTNKVC) [inline src/resources/views/components/mail/grid.blade.php]: This wont work as the component will simply use the default class, which in this case is 'mobile'. In order for this to work, you would need to merge the default class with whatever is passed when calling a component:

```blade
<td {{ $four->attributes->merge(['class' => 'mobile']) }} align="center" valign="top" >
    {{ $four }}
</td>
```

Comment (BJNSTNKVC) [reply src/resources/views/components/mail/grid.blade.php]: @Korko Sorry for the late response, I was away. First of all, thank you for taking an intereset and your time!

I like the idea, however implementation needs to be tweaked a little bit in order for it to work. I left some comments/reviews detailing my take on the issue.

File: src/resources/views/components/mail/grid.blade.php (hunk 1 above)
Comment (Korko) [reply src/resources/views/components/mail/grid.blade.php]: It does work if you want to inject something else than class, like for example raw style. But I get your point with class, yeah.

**Review by BJNSTNKVC — APPROVED** [review]
Approved, thanks!

**Pull request merged**